### PR TITLE
UI - Infantry speed indictator

### PR DESCRIPTION
**Pull request by veteran29**
**When merged this pull request will:**
- title

ace_speed_indicator

### CODE REVIEW

File: addons/advanced_fatigue/script_component.hpp [hunk 1]
```diff
@@ -14,6 +14,7 @@
     #define DEBUG_SETTINGS DEBUG_SETTINGS_ADVANCED_FATIGUE
 #endif
 
+#include "\a3\ui_f\hpp\defineCommonGrids.inc"
```
Comment (jonpas) [inline addons/advanced_fatigue/script_component.hpp:17]: Should be below macros include.

File: addons/advanced_fatigue/RscInGameUI.hpp [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+class RscInGameUI {
+    class RscStanceInfo {
+        controls[] += {QGVAR(speedIndicator)};
+        class GVAR(speedIndicator): RscPictureKeepAspect {
+            onLoad = QUOTE(uiNamespace setVariable [ARR_2(QQGVAR(speedIndicator),_this select 0)]);
+            x = IGUI_GRID_STANCE_X + IGUI_GRID_STANCE_WAbs / 2 - SPEED_INDICATOR_W / 2;
+            y = IGUI_GRID_STANCE_Y + IGUI_GRID_STANCE_HAbs - SPEED_INDICATOR_Y_OFFSET);
+            w = SPEED_INDICATOR_W;
+            h = SPEED_INDICATOR_H;
+
```
Comment (jonpas) [inline addons/advanced_fatigue/RscInGameUI.hpp:15]:
```suggestion
```

**Review by jonpas — CHANGES_REQUESTED** [review]

File: addons/advanced_fatigue/script_component.hpp (hunk 1 above)
Comment (veteran29) [reply addons/advanced_fatigue/script_component.hpp:17]: Copied from `medical_feedback` ;P
https://github.com/acemod/ACE3/blob/master/addons/medical_feedback/script_component.hpp#L17

**Review by jonpas — APPROVED** [review]

File: addons/advanced_fatigue/RscInGameUI.hpp [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+#define SPEED_INDICATOR_Y_OFFSET (IGUI_GRID_STANCE_HAbs / 4.25)
+
+class RscPictureKeepAspect;
+class RscInGameUI {
+    class RscStanceInfo {
+        controls[] += {QGVAR(speedIndicator)};
+        class GVAR(speedIndicator): RscPictureKeepAspect {
+            onLoad = QUOTE(uiNamespace setVariable [ARR_2(QQGVAR(speedIndicator),_this select 0)]);
+            x = IGUI_GRID_STANCE_X + IGUI_GRID_STANCE_WAbs / 2 - SPEED_INDICATOR_W / 2;
+            y = IGUI_GRID_STANCE_Y + IGUI_GRID_STANCE_HAbs - SPEED_INDICATOR_Y_OFFSET);
```
Comment (PabstMirror) [inline addons/advanced_fatigue/RscInGameUI.hpp:12]:
```suggestion
            y = IGUI_GRID_STANCE_Y + IGUI_GRID_STANCE_HAbs - SPEED_INDICATOR_Y_OFFSET;
```

**Comment by Drofseh** [comment]
I would actually really like this not to be in Advanced Fatigue, a lot of units don't use AF and this is still valuable info for the regular stamina system

**Comment by veteran29** [comment]
> I would actually really like this not to be in Advanced Fatigue, a lot of units don't use AF and this is still valuable info for the regular stamina system

I have no intention (and motivation) to separate this from AF, feel free to do so in a PR after this gets merged.

**Comment by Drofseh** [comment]
Yah I really think this should be in `UI`, it's not even related to fatigue

**Comment by BaerMitUmlaut** [comment]
I don't see any reason why this needs to be bound to AF, since it doesn't access any AF functionality.

**Comment by veteran29** [comment]
My main idea was that you do not need careful stamina management in vanilla system and such indicator will be not useful/needed in there.

**Comment by veteran29** [comment]
Found some time to move it to UI.

File: addons/ui/functions/fnc_handleSpeedIndicator.sqf [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+ * Handles visual changes of the speed indicator.
+ *
+ * Arguments:
+ * Current animation state <STRING>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * ["amovpercmtacslowwrfldf_ver2"] call ace_advanced_fatigue_fnc_handleSpeedIndicator
```
Comment (PabstMirror) [inline addons/ui/functions/fnc_handleSpeedIndicator.sqf:13]:
```suggestion
 * ["amovpercmtacslowwrfldf_ver2"] call ace_ui_fnc_handleSpeedIndicator
```

**Review by PabstMirror — APPROVED** [review]

**Review by jonpas — APPROVED** [review]

**Pull request merged**